clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -79,6 +82,12 @@
     Properties properties = configuration.getProperties();
     this.connectionState = new ConnectionState(clusterUri, configuration.getTimeouts(),
             entityIdentifier, properties, configuration);
+    this.connectionState.onConnectionRecovery(() -> connectionRecoveryListeners.forEach(Runnable::run));
+  }
+
+  @Override
+  public void onConnectionRecovery(Runnable runnable) {
```

Comment:
Same issue. It should be a `addConnectionRecoveryListener`. Or `addConnectionRecoveryListered`